### PR TITLE
Test for setting email notifier SMTP password

### PR DESCRIPTION
**Pull request by XianglongLuo**
Problem: Notifier email configure return plaintext password
Solution: update type to password
Related PR:https://github.com/rancher/types/pull/1051
Issue: https://github.com/rancher/rancher/issues/24335

### CODE REVIEW

File: tests/integration/suite/test_notifier.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    notifier = client.reload(notifier)
+    assert notifier["smtpConfig"]["username"] == "test2"
+
+    validate_password_exists_in_crd(admin_mc, password)
+
+
+def validate_password_exists_in_crd(admin_mc, password):
+    cluster, client = cluster_and_client('local', admin_mc.client)
+    cc = ClusterContext(admin_mc, cluster, client)
+    create_kubeconfig(cc.cluster)
```
Comment (kinarashah) [inline tests/integration/suite/test_notifier.py]: we don't need to create kubeconfig to check the output of password, we can use k8s client, 
for example, crdClient used here: https://github.com/rancher/rancher/blob/master/tests/integration/suite/test_password_store.py#L175

File: tests/integration/suite/test_notifier.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
+from kubernetes.client import CustomObjectsApi
+from .common import random_str
+
+group = 'management.cattle.io'
+version = 'v3'
+namespace = 'local'
+plural = 'notifiers'
```
Comment (dramich) [inline tests/integration/suite/test_notifier.py]: This set of things should be held in a dict, preferably inside of `verifySmtpPassword`, as they are specific to that function and having them global can cause potential issues in the future.

File: tests/integration/suite/test_notifier.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+clusterId = "local"
```
Comment (dramich) [inline tests/integration/suite/test_notifier.py]: This isn't used anywhere

File: tests/integration/suite/test_notifier.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    # Test the password in crd remains the same value after updating username
+    verifySmtpPassword(crdClient, ns, name, newPassword)
+
+
+def verifySmtpPassword(crdClient, ns, name, Password):
+    k8notifier = crdClient.get_namespaced_custom_object(
+        group, version, namespace, plural, name)
+    smtpPassword = k8notifier['spec']['smtpConfig']['password']
+
+    assert smtpPassword is not None
```
Comment (dramich) [inline tests/integration/suite/test_notifier.py]: The None check isn't needed

File: tests/integration/suite/test_notifier.py [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    verifySmtpPassword(crdClient, ns, name, newPassword)
+    # Test updating field non-password related
+    notifier = client.update(notifier, smtpConfig={"username": "test2"})
+    notifier = client.reload(notifier)
+    assert notifier["smtpConfig"]["username"] == "test2"
+    # Test the password in crd remains the same value after updating username
+    verifySmtpPassword(crdClient, ns, name, newPassword)
+
+
+def verifySmtpPassword(crdClient, ns, name, Password):
```
Comment (dramich) [inline tests/integration/suite/test_notifier.py]: `Password` should not be capitalized

**Comment by dramich** [comment]
There are multiple places that camelCase is being used for vars and func declarations, this should follow our convention of using snake case. https://www.python.org/dev/peps/pep-0008/#function-and-variable-names

**Comment by XianglongLuo** [comment]
updated,Thanks.

**Pull request merged**